modules/hibp/settings.go

```diff
@@ -44,9 +44,10 @@ func NewSettingsFromYAML(name string, ymlConfig *config.Config, globalConfig *co
 		globalConfig,
 		"https://haveibeenpwned.com/api",
 		"",
-		&settings.apiKey,
 		nil,
+		&settings.apiKey,
```

Comment:
This is the reason for the comment at https://github.com/sam-github/wtf/blob/bded9475203da9f0789ecfc7db6c426aa7906280/cfg/secrets.go#L42

Having multiple positional args with the same type signature leads to this confusion.

I thought about using the pattern where a series of calls defines the options. The usage would be the more verbose but more clear by avoiding positional args.
```
cfg.ConfigureSecret(globalConfig, "https://haveibeenpwned.com/api")
  //  not needed here: .ServiceDefault("")
  .Secret(&settings.apiKey)
// not needed here: .Username(&...) 
  .Load()
```
I guess this would work, too:
```
cfg.ConfigureSecret(globalConfig, &SecretOptions{
   Service: "https://haveibeenpwned.com/api",
   Secret: &settings.apiKey,
   // Omitted args: 
  //  ServiceDefault:
   // Username: 
 })
```

thoughts, @senorprogrammer ?

Reply:
What do you think about keeping the inlined params but rather than have them be `string`, create custom type definitions that enforce them? Something like:

```
package main

import (
	"fmt"
)

type Username string
type Secret string

func main() {
	uname := Username("user")
	pw := Secret("pass")
	ConfigureSecret(uname, pw)
}

func ConfigureSecret(a Username, b Secret) {
	fmt.Println(a)
	fmt.Println(b)
}
```

https://play.golang.org/p/60lCbdbpanP

Reply:
Its a nice light-weight but explicit approach, but would require changing the type definition in every config structure that wanted to use this: https://github.com/wtfutil/wtf/blob/a19c9c3e330d6da9bd1210f77e9476abcc945efd/modules/circleci/settings.go#L18 would have to change to
```
apiKey cfg.Secret `help:"Your CircleCI API token."`
```
and the apiKey settings below at https://github.com/wtfutil/wtf/blob/a19c9c3e330d6da9bd1210f77e9476abcc945efd/modules/circleci/settings.go#L26 would need to change to something like `ymlConfig.Secret("apiKey")` to fix the type mismatch, which suggests that ymlConfig should just fetch the secret... which isn't such a bad idea, except that fetching the secret requires knowing which API its for (not for circleci, but for modules with multiple possible API endpoints, like github enterprise), and that leaves us in a jam, having a config setting depend on the configured value of another setting.